File: demo/csharp/positions/positions.cs
using System;
using CosineKitty;
using demo_helper;

namespace positions
{
    class Program
    {
        static int Main(string[] args)
        {
            var bodies = new Body[]
            {
                Body.Sun, Body.Moon, Body.Mercury, Body.Venus, Body.Mars,
                Body.Jupiter, Body.Saturn, Body.Uranus, Body.Neptune, Body.Pluto
            };

            Observer observer;
            AstroTime time;
            DemoHelper.ParseArgs("positions", args, out observer, out time);
            Console.WriteLine("UTC date = {0}", time);
            Console.WriteLine();
            Console.WriteLine("BODY           RA      DEC       AZ      ALT");
            foreach (Body body in bodies)
            {
                Equatorial equ_2000 = Astronomy.Equator(body, time, observer, EquatorEpoch.J2000, Aberration.Corrected);
                Equatorial equ_ofdate = Astronomy.Equator(body, time, observer, EquatorEpoch.OfDate, Aberration.Corrected);
                Topocentric hor = Astronomy.Horizon(time, observer, equ_ofdate.ra, equ_ofdate.dec, Refraction.Normal);
                Console.WriteLine("{0,-8} {1,8:0.00} {2,8:0.00} {3,8:0.00} {4,8:0.00}", body, equ_2000.ra, equ_2000.dec, hor.azimuth, hor.altitude);
            }

            return 0;
        }
    }
}

File: demo/java/src/main/java/io/github/cosinekitty/astronomy/demo/Positions.java
package io.github.cosinekitty.astronomy.demo;
import io.github.cosinekitty.astronomy.*;

public class Positions {
    private static final Body[] bodyList = new Body[] {
        Body.Sun, Body.Moon, Body.Mercury, Body.Venus, Body.Mars,
        Body.Jupiter, Body.Saturn, Body.Uranus, Body.Neptune, Body.Pluto
    };

    /**
     * Print a table of solar system body positions in equatorial and horizontal coordinates.
     *
     * @param observer
     * The geographic location for which to calculate positions.
     *
     * @param time
     * The date and time for which to calculate positions.
     */
    public static int run(Observer observer, Time time) {
        System.out.printf("UTC date = %s%n", time);
        System.out.println();
        System.out.println("BODY           RA      DEC       AZ      ALT");
        for (Body body : bodyList) {
            Equatorial equ_2000 = Astronomy.equator(body, time, observer, EquatorEpoch.J2000, Aberration.Corrected);
            Equatorial equ_ofdate = Astronomy.equator(body, time, observer, EquatorEpoch.OfDate, Aberration.Corrected);
            Topocentric hor = Astronomy.horizon(time, observer, equ_ofdate.getRa(), equ_ofdate.getDec(), Refraction.Normal);
            System.out.printf("%-8s %8.2f %8.2f %8.2f %8.2f%n", body, equ_2000.getRa(), equ_2000.getDec(), hor.getAzimuth(), hor.getAltitude());
        }
        return 0;
    }
}
